edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Service/RegistryProxyMiddleware.cs

```diff
@@ -0,0 +1,119 @@
+                {
+                    await this.CopyProxyResponseAsync(context, responseMessage);
+                }
+            }
+            else
+            {
+                await this.nextMiddleware(context);
+            }
+        }
+
+        internal HttpRequestMessage CreateProxyRequestMessage(HttpContext context, Uri destinationUri)
+        {
+            var request = context.Request;
+            var requestMessage = new HttpRequestMessage();
+            var requestMethod = request.Method;
+
+            if (!HttpMethods.IsGet(requestMethod) &&
+                !HttpMethods.IsHead(requestMethod) &&
+                !HttpMethods.IsDelete(requestMethod) &&
+                !HttpMethods.IsTrace(requestMethod))
+            {
+                var streamContent = new StreamContent(request.Body);
+                requestMessage.Content = streamContent;
+            }
+
+            foreach (var header in request.Headers)
+            {
+                if (!requestMessage.Headers.TryAddWithoutValidation(header.Key, header.Value.ToArray()) && requestMessage.Content != null)
+                {
+                    requestMessage.Content?.Headers.TryAddWithoutValidation(header.Key, header.Value.ToArray());
+                }
+            }
+
+            requestMessage.RequestUri = destinationUri;
+            requestMessage.Headers.Host = destinationUri.Authority;
+            requestMessage.Method = new HttpMethod(requestMethod);
+
+            return requestMessage;
+        }
+
+        internal async Task CopyProxyResponseAsync(HttpContext context, HttpResponseMessage responseMessage)
+        {
+            if (responseMessage == null)
+            {
+                throw new ArgumentNullException(nameof(responseMessage));
+            }
+
+            HttpResponse response = context.Response;
+
+            response.StatusCode = (int)responseMessage.StatusCode;
+            foreach (var header in responseMessage.Headers)
+            {
+                response.Headers[header.Key] = header.Value.ToArray();
+            }
+
+            foreach (var header in responseMessage.Content.Headers)
+            {
+                response.Headers[header.Key] = header.Value.ToArray();
+            }
+
+            // SendAsync removes chunking from the response. This removes the header so it doesn't expect a chunked response.
+            response.Headers.Remove("transfer-encoding");
+
+            await responseMessage.Content.CopyToAsync(response.Body);
+        }
+
+        internal Uri BuildDestinationUri(HttpContext context)
+        {
+            HttpRequest request = context.Request;
+
+            // This pattern is used to match registry API for create, update, get, delete of a module, and list modules.
+            Regex registryApiUriPattern = new Regex(@"\/devices\/.+\/modules(\/.+)?$");
```

Comment:
What is the pattern here? 
just /devices/deviceId/modules/moduleId ?

Reply:
You can find all references of iotedged calling to registry api in this file, https://github.com/Azure/iotedge/blob/4425533aed41cd3eed4780612af214a075b51093/edgelet/iothubservice/src/device.rs#L91.  This pattern is for GET, PUT, DELETE of /devices/{deviceId}/modules/{moduleId} and GET of /devices/{deviceId}/modules.